nav-menu/src/components/Navigation.js

```diff
@@ -0,0 +1,58 @@
+import React, {Component} from 'react';
+import Navbar from './Navbar';
+import Jumbo from './Jumbo';
+import Registr from './Registr';
+import LogIn from './LogIn';
+import Page from './Page';
+
+class Navigation extends Component{
+    constructor(){
+        super();
+        this.state ={
+            currentPage: 'home'
+        }
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(page){
+        this.setState({
+            currentPage: page
+        });
+    }
+
+    render(){
+        let jumboComponent;
+        if(this.state.currentPage == 'Home'){
```

Comment:
a lot of if... statements here. I guess we may create something like Component Factory.
For example
```
function getComponent( page ) {
  let component;
   switch( page) {
     case 'home':
           component = <Jumbo>;
           break;
     case 'register'
.....
   }
}
```
and usage
```
let component = getComponent( this.state.currentPage );
```
this will reduce amount of code you have and will make it more readable.

Reply:
@wizzardio code has been improved